google/docs/update_doc.py

```diff
@@ -0,0 +1,145 @@
+import sys
+import os
+import json
+
+import markdown
+from bs4 import BeautifulSoup
+
+from auth import client
+from id import extract_file_id
+
+def markdown_to_google_doc_requests(markdown_content):
+    # Convert markdown content to HTML
+    html_content = markdown.markdown(markdown_content)
+
+    # Parse the HTML content using BeautifulSoup
+    soup = BeautifulSoup(html_content, 'html.parser')
+
+    requests = []
+
+    # Track the current index for insertion
+    current_index = 1
+
+    # Helper to add text with styles
+    def add_text_request(text, bold=False, italic=False, underline=False, link=None):
+        nonlocal current_index
+        text_style = {}
+        # Add styles only if explicitly specified
+        if bold:
+            text_style['bold'] = True
+        if italic:
+            text_style['italic'] = True
+        if underline:
+            text_style['underline'] = True
+        if link:
+            text_style['link'] = {"url": link}
+
+        # Add text insertion request
+        text_length = len(text)
+        requests.append({
+            "insertText": {
+                "location": {"index": current_index},
+                "text": text
+            }
+        })
+
+        # Add styling request only if any styles are present
+        if text_style:
+            requests.append({
+                "updateTextStyle": {
+                    "range": {
+                        "startIndex": current_index,
+                        "endIndex": current_index + text_length
+                    },
+                    "textStyle": text_style,
+                    "fields": ",".join(text_style.keys())
+                }
+            })
+
+        # Update the current index to account for the added text
+        current_index += text_length
+
+    # Process elements in the markdown
+    for element in soup.contents:
+        if element.name in ['p']:
+            add_text_request(element.get_text() + "\n")
+        elif element.name in ['h1', 'h2', 'h3']:
+            # Apply bold style for headers
+            add_text_request(element.get_text() + "\n", bold=True)
```

Comment:
This needs to change a bit. It's converting headers to bold instead of the respective google docs `hX` size.